SQL-based implementation of the executive report.

mreid-moz opened the pull request.

mreid-moz committed:
Initial import of sql-based exec report runner.

mreid-moz committed:
Stop creating views, build subselects inline.

Instead of creating "this_week" views, just insert the select list
inline.

mreid-moz committed:
Add monthly mode.

mreid-moz committed:
Run the executive report against Redshift.

mreid-moz committed:
Add an option to skip missing tables.

The '--check-tables' option allows the report to be run against an
incomplete date range, and emits a bunch of ugly warning messages
that the output will be incomplete.

mreid-moz committed:
Remove debug prints.

mreid-moz committed:
Fix the this_month/last_month logic

mreid-moz committed:
Remove a superfluous print stmt.

mreid-moz committed:
Update to fetch DB_URL from metadata

mreid-moz committed:
Print error msg if 'reformat' script is missing.

Can't just wget it since it lives in a private repo :(

mreid-moz committed:
Add missing psycopg2 prerequisite.

mreid-moz committed:
Update total_records column (actives + inactives)

mreid-moz committed:
Fix "inactives" counting, a typo and some comments

Query was trying to get the per-client data for inactives from the
current period instead of the previous period, which by definition
wouldn't contain any info for the inactives.

mreid-moz committed:
Properly fetch DB credentials, compare csv header.

Fetch the DB credentials from the metadata bucket. Check the csv
header for the current run against the "overall" data, emit a
warning if they differ.

mreid-moz committed:
Check the tables and emit a warning if missing.

mreid-moz committed:
Clean up SQL

mreid-moz committed:
Fix a typo

mreid-moz committed:
Add the 'acl' argument for cross-iam annoyance.

mreid-moz committed:
Change the defaults for reporting period.

Since the Redshift data can lag behind a bit, we want to ensure
that the report can be run a few days after the end of the reporting
period.

Instead of running for the period ending "yesterday", we use the
most recent completed period.

trink commented on reports/executive_summary/README.txt:
```diff
@@ -5,13 +5,25 @@ To Deploy
     Day of Month:          n/a (1)
-    Time of Day (UTC):     2pm
+    Time of Day (UTC):     10am
     Job Timeout (minutes): 120
```
How quickly does the weekly job run?

mreid-moz replied:
I'm not sure with the change to fewer, larger machines in the cluster. I'll check again. Meanwhile, I'll bump this up to 300 for weekly and 600 for monthly just to be safe.

trink commented on reports/executive_summary/run.sh:
```diff
@@ -1,42 +1,107 @@
+OVERALL="v4-${MODE}.csv"
+DASHBOARD_S3="s3://net-mozaws-prod-metrics-data/firefox-executive-dashboard"
+echo "Fetching previous state from $OVERALL..."
+aws s3 cp "$DASHBOARD_S3/$OVERALL" ./
```
We should check the return codes on all these network operations (I have had them fail before).  This could assume no previous state when there actually was one.

mreid-moz replied:
Good idea. I will add that.

mreid-moz replied:
Actually, I think some helper code for fetching state, metadata, credentials, etc could be in order...

mreid-moz committed:
Much more conservative estimates for timeouts.

mreid-moz committed:
Check awscli return codes.

Per review feedback, check that the aws transfers succeeded before
we run the report.
